R/QFeatures-constructors.R

```diff
@@ -1,85 +1,233 @@
+##' - The multi-set case will generate a `QFeatures` object with
+##'   multiple [SummarizedExperiment] sets, resulting from splitting
+##'   the input table.
 ##'
-##' @param fnames An optional `character(1)` or `numeric(1)`
-##'     indicating the column to be used as feature names.
+##' @param assayData A `data.frame`, or any object that can be coerced
+##'     to a `data.frame`, holding the quantitative assay. For
+##'     `readSummarizedExperiment()`, this can also be a
+##'     `character(1)` to a filename.
 ##'
-##' @param ... Further arguments that can be passed on to `read.csv`
-##'     except `stringsAsFactors`, which is always `FALSE`.
+##' @param colAnnotation The type of this parameter will define
+##'     whether the resulting `QFeaures` object will contain a single
+##'     or multiple sets.
 ##'
-##' @param name An `character(1)` to name assay in the `QFeatures`
-##'     object. If not set, `features` is used.
+##'     For the single-set case, a `numeric` indicating the indices of
+##'     the columns to be used as expression values, or a `character`
+##'     indicating the names of the columns.
 ##'
-##' @return An instance of class [QFeatures] or [SummarizedExperiment].
+##'     For the multi-set case, a `data.frame` or any object that can
+##'     be coerced to a `data.frame`. It is expected to contain all
+##'     the sample meta information. Required fields are the
+##'     acquisition batch (given by `batchCol`) and the acquisition
+##'     channel within the batch (e.g. TMT channel, given by
+##'     `channelCol`). Additional fields (e.g. sample type,
+##'     acquisition date,...) are allowed and will be stored as sample
+##'     meta data.
 ##'
-##' @author Laurent Gatto
+##' @param fnames For the single- and multi-set cases, an optional
+##'     `character(1)` or `numeric(1)` indicating the column to be
+##'     used as feature names. If missing, in the latter case, the
+##'     rows are named PSM1, PSM2, PSM3, ... Note that rownames must
+##'     be unique in `QFeatures` sets.
 ##'
-##' @describeIn readQFeatures See description.
+##' @param name For the single-set case, an optional `character(1)` to
+##'     name the set in the `QFeatures` object. If not set, `psms`
+##'     is used.
+##'
+##' @param batchCol For the multi-set case, a `numeric(1)` or
+##'     `character(1)` pointing to the column of `assayData` and
+##'     `colAnnotation` that contain the batch names. Make sure that
+##'     the column name in both table are either identical and
+##'     syntactically valid (if you supply a `character`) or have the
+##'     same index (if you supply a `numeric`). Note that characters
+##'     can be converted to syntactically valid names using
+##'     `make.names`
+##'
+##' @param channelCol For the multi-set case, a `numeric(1)` or
+##'     `character(1)` pointing to the column of `colData` that
+##'     contains the column names of the quantitative data in
+##'     `assayData` (see Example).
```

Comment:
Maybe clarify that `channelCol` is useful for multiplexed experiments such as mTRAQ, TMT (SILAC?)